plugins/intel-usb4/fu-intel-usb4-device.c

```diff
@@ -589,7 +589,7 @@ fu_intel_usb4_device_init(FuIntelUsb4Device *self)
 	fu_device_add_protocol(FU_DEVICE(self), "com.intel.thunderbolt");
 	fu_device_add_flag(FU_DEVICE(self), FWUPD_DEVICE_FLAG_UPDATABLE);
 	fu_device_add_flag(FU_DEVICE(self), FWUPD_DEVICE_FLAG_SIGNED_PAYLOAD);
-	fu_device_set_version_format(FU_DEVICE(self), FWUPD_VERSION_FORMAT_PAIR);
+	fu_device_set_version_format(FU_DEVICE(self), FWUPD_VERSION_FORMAT_BCD);
```

Comment:
I'm a bit confused why this is needed -- isn't bcd a subset of pair?

Reply:
Was it wrong before?

Reply:
The raw version conversion is a bit different between BCD and others.

For example, a raw version say 16842888 in `dec`, when converting to `BCD` with `BE` it is resulted in `88.00.01.01`.  

Go with `QUAD` that will become `136.0.0.1`.

IIRC thunderbolt version has this difference between PAIR and BCD.

Reply:
Maybe this is an issue with version conversion?

Reply:
Given `guint32 val = 0x88000101`, current output with `QUAD` from [fu_version_from_uint32](https://github.com/fwupd/fwupd/blob/aae6101a428a0ce54f83d92af7d4025b3e197607/libfwupdplugin/fu-version-common.c#L87) is 136.0.0.1.

I'm actually looking to translate this val to 88.0.1.1, which should go with `%x` instead `%u`.  What version format should I use?

For now, am using `BCD` because it gives the hex value of desire. But, the conversion complains that 88.0.1.1 isn't a valid BCD.